### PR TITLE
Increase test coverage with substitute_links() and ignoring demos

### PR DESCRIPTION
**Pull request by bfolie**
Write tests to cover every line of `taurus.util.impl.substitute_links()` and update config file so that `demo/` is ignored. Bumps coverage to 94%.

### CODE REVIEW

File: taurus/util/tests/test_substitute_links.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    meas = MeasurementRun("A measurement", material=mat,
+                          uids={native_id: str(uuid4()), other_id: str(uuid4())})
+
+    # Turn the material pointer into a LinkByUID using native_id
+    substitute_links(meas, native_uid=native_id)
+    assert meas.material == LinkByUID.from_entity(mat, name=native_id)
+
+    # Put the measurement into a list and convert that into a LinkByUID using native_id
+    measurements_list = [meas]
+    substitute_links(measurements_list)
```
Comment (maxhutch) [inline taurus/util/tests/test_substitute_links.py]:
```suggestion
    substitute_links(measurements_list, native_uid=native_id)
```

File: taurus/util/tests/test_substitute_links.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    with pytest.raises(ValueError):
+        substitute_links({mat: meas}), "substitute_links should fail if objects don't have uids"
+    with pytest.raises(ValueError):
+        substitute_links({meas: mat}), "substitute_links should fail if objects don't have uids"
+
+
+def test_native_id_substitution():
+    """Test that the native id gets serialized, when specified."""
+    native_id = 'id1'
+    other_id = 'id2'
```
Comment (maxhutch) [inline taurus/util/tests/test_substitute_links.py]: You might want to add more other_ids so the probability of randomly selecting the `native_id` is lower.

File: taurus/util/tests/test_substitute_links.py (hunk 1 above)
Comment (bfolie) [reply taurus/util/tests/test_substitute_links.py]: Good catch. Fixed

File: taurus/util/tests/test_substitute_links.py (hunk 2 above)
Comment (bfolie) [reply taurus/util/tests/test_substitute_links.py]: Done

**Pull request merged**